hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -615,6 +632,24 @@ private Action handleAddUpdateActionStatus(final JpaActionStatus actionStatus, f
         return savedAction;
     }
 
+    private String handleDownloadedActionStatus(final JpaActionStatus actionStatus, final JpaAction action) {
+        // information status entry - check for a potential DOS attack
+        assertActionStatusQuota(action);
+        assertActionStatusMessageQuota(actionStatus);
+
+        return isDownloadOnly(action) ? handleDownloadedActionStatusForDownloadOnlyAction(action) : null;
```

Comment:
Is it intended that the DOWNLOADED status is ignored for actions that are not download only?

Reply:
We currently do not have a device state holder for downloaded distributionSets. In this case an intermediate "DOWNLOADED" update does not change the action state. For non-Download_only actions, a "DOWNLOADED" actionStatusUpdate is still mapped to ActionStatus.RUNNING, so nothing to do here.